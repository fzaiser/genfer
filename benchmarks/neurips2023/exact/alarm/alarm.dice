let earthquake  = flip 0.0001 in
let burglary  = flip 0.001 in
let alarm  = earthquake || burglary in
let phoneWorking  = if earthquake then flip 0.7 else flip 0.99 in
let maryWakes  = if alarm then
    (if earthquake then flip 0.8 else flip 0.6)
    else flip 0.2 in
let called  = (maryWakes && phoneWorking) in
let tmp = observe called in
burglary
