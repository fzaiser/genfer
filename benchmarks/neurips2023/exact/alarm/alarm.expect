Support is a subset of: {0, ..., 1}

Computing moments...
Total measure:             Z = 0.1984321604
Expected value:            E = 0.002993449241305544
Standard deviation:        Ïƒ = 0.05463047229289961
Variance:                  V = 0.002984488502945272
Skewness (3rd std moment): S = 18.19521340009689
Kurtosis (4th std moment): K = 332.0657906750655

Computing probabilities up to 2...
Unnormalized: p(0)     = 0.1978381638
Normalized:   p(0) / Z = 0.9970065507586944
Unnormalized: p(1)     = 0.0005939966
Normalized:   p(1) / Z = 0.002993449241305544
Unnormalized: p(n)     <= 0.0 for all n >= 2
Normalized:   p(n) / Z <= 0.0 for all n >= 2
