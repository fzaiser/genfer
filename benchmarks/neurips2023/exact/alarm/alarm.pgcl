nat earthquake;
nat burglary;
nat alarm;
nat phoneWorking;
nat maryWakes;
nat called;

earthquake := bernoulli(1/10000);
burglary := bernoulli(1/1000);
if (earthquake = 1 || burglary = 1) {
    alarm := 1
} else {skip}

if (earthquake = 1) {
    phoneWorking := bernoulli(7/10);
} else {
    phoneWorking := bernoulli(99/100);
}

if (alarm = 1) {
    if (earthquake = 1) {
        maryWakes := bernoulli(4/5);
    }else{
        maryWakes := bernoulli(3/5);
    }
} else {
    maryWakes := bernoulli(1/5);
}

if (maryWakes = 1 & phoneWorking = 1){
    called := 1;
} else {skip}
observe(called = 1);

?Pr[burglary];
