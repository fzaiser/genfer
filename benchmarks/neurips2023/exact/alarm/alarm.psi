// flags: --dp
def main(){ // isBurglary
    earthquake := flip(1/10000);
    burglary := flip(1/1000);
    alarm := earthquake || burglary;

    phoneWorking := 0;
    if earthquake{
        phoneWorking = flip(7/10);
    }else{
        phoneWorking = flip(99/100);
    }

    maryWakes := 0;
    if alarm{
        if earthquake{
            maryWakes = flip(4/5);
        }else{
            maryWakes = flip(3/5);
        }
    }else{
        maryWakes = flip(1/5);
    }

    called := maryWakes && phoneWorking;
    observe(called);

    return burglary;
}
