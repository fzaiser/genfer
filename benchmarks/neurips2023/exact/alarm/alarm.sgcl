earthquake ~ Bernoulli(0.0001);
burglary ~ Bernoulli(0.001);
if earthquake = 1 or burglary = 1 {
    alarm := 1;
}

if (earthquake=1){
    phoneWorking ~ Bernoulli(0.7);
} else {
    phoneWorking ~ Bernoulli(0.99);
}

if alarm = 1 {
    if earthquake = 1 {
        maryWakes ~ Bernoulli(0.8);
    } else {
        maryWakes ~ Bernoulli(0.6);
    }
} else {
    maryWakes ~ Bernoulli(0.2);
}

if maryWakes = 1 and phoneWorking = 1 {
    called := 1;
}
observe called=1;

return burglary;
