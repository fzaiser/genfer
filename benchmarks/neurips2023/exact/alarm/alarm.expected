2969983/992160802
0.00299344
